lib/ex_format.ex

```diff
@@ -7,16 +13,60 @@ defmodule ExFormat do
     Lines,
   }
 
-  def format(string) do
-    initialize_stores(string)
-    ast = AST.initialize_ast(string)
-    state = %State{}
+  @doc ~S"""
+  Formats the given code string.
+
+  It returns a tuple containing the formatted code string, as well as
+  information about whether the code string was formatted successfully or not.
+
+  ## Examples
+      iex> ExFormat.format("   quote(do:  foo.bar(1,2,     3))")
+      {:ok, "quote do: foo.bar(1, 2, 3)\\n"}
+
+      iex> ExFormat.format("   quote(do:foo.bar(1,2,     3))")
+      {:error, "nofile:1: keyword argument must be followed by space after: do:"}
+  """
+  @spec format(String.t) :: {:ok, String.t} | {:error, String.t}
+  def format(code_string) do
```

Comment:
Okay, I'll just revert back to the original code which doesn't do rescue and remove `format!/1`